### ISSUE TITLE
Import problem with Geocaching Australia GPX

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#5301]
From support mail:

This is a GPX file from Geocaching Australia which does not seem to parse correctly. When I import it I get the waypoint code and that’s it. The type and size and description etc, etc, are all not there when I open c:geo. The co-ordinates are there however which is a little strange if the other data is not importing. The file attached looks like a valid GPX file and validates against the appropriate schema, but doesn’t import all of the data.

My device is a Samsung Galaxy Tab 4 running Android 4.4.2

GPX:
https://www.dropbox.com/s/57xzo6nyyfqupjt/ga7254.gpx?dl=0

Scrrenshot:
https://www.dropbox.com/s/1d6v94oxti27z50/Screenshot_2015-11-26-11-57-11.png

**Comment by Bananeweizen** [comment]
This isn't a bug in cgeo, but rather the geocaching.com.au people not correctly understanding what XML schemas are meant for. That GPX file declares that most of the XML elements in the GPX file are described by a schema located at the geocaching.com.au server, see this line:

    <groundspeak:cache id="98007254" available="True" archived="False" xmlns:groundspeak="http://www.geocaching.com.au/geocache/1/0/1">

That means all the elements of the form groundspeak:something that come afterwards are described by that linked schema. And the problem is, cgeo doesn't know what this schema describes, as cgeo only knows the XML schema for geocaches, which is the one located at http://www.groundspeak.com/cache/1/0/1.

Basically, it was plain wrong by geocaching.com.au people to replace all occurences of "geocaching.com" in GPX exports by their own domain name. Even though the GPX file looks like it contains geocaches, it actually doesn't. It contains something else, that cgeo doesn't know about.

And the coordinates are still recognized by cgeo because are described by _another_ schema, which is the topographix schema. And the URL of that has not been changed (because it doesn't contain geocaching.com in the name), so cgeo agrees on knowing those elements.

**Comment by Lineflyer** [comment]
Given the recent mails on support I imagine they only recently changed this and therefore broke it.
I will try to see when their site is online again, but anyone else: Feel free to try as my time might be limited.

**Comment by Bananeweizen** [comment]
...waiting for account being confirmed...

**Comment by Bananeweizen** [comment]
posted at their forum

**Comment by CraigRat** [comment]
G'day,

Nothing has changed with our export since 2011!

Other than the location of the XSD our 1.0.1 version uses groundspeak namespaces etc and works with most things. Looks like c:geo isn't one of them.

That said, happy to work with the c:geo devs.

And just a small thing: 
>c:geo only knows the XML schema for geocaches

This is not 'the' schema for geocaches. It is groundspeaks schema for geocaches. Slight but important distinction.

Cheers

CraigRat

GCA Developer.

**Comment by caughtatwork** [comment]
Hi there.

The file is valid and parses correctly according to any number of XML validators. It also parses correctly within the Garmin GPS units.

It would seem that cgeo is not parsing according to the defined schema but the Groundspeak schema which will cause errors. The file contains the XSD reference, so this should be used the perform the parsing and not the Groundspeak schema.

If the cgeo team can parse the data according to the XSD indicated in the file, the same way that say, GSAK does, then they will find the file is perfectly valid.

The schema defined by Groundspeak is really their own schema and not a geocaching schema. There is no such thing as a geocaching schema.

Regards.
caughtatwork.

**Comment by rsudev** [comment]
@CraigRat Unfortunately do you not use the groundpseak namespace (anymore). The namespace is defined by the url you supply, not the alias you choose...

**Comment by Bananeweizen** [comment]
I'm giving up here. I'm not interested anymore in fixing this.

**Comment by Bananeweizen** [comment]
cgeo is parsing exactly according to the schemas defined in the XML, however it does not interpret values of unknown schemas. How should it? Just by linking to a shopping list schema in the GPX file doesn't make cgeo an app that understands shopping lists. And that's what you are trying by referencing your own schema.

**Comment by samueltardieu** [comment]
> It would seem that cgeo is not parsing according to the defined schema but the Groundspeak schema which will cause errors. The file contains the XSD reference, so this should be used the perform the parsing and not the Groundspeak schema.

> If the cgeo team can parse the data according to the XSD indicated in the file, the same way that say, GSAK does, then they will find the file is perfectly valid.

You don't seem to understand the issue. As @rsudev said, a namespace is uniquely identified by its URI, independently of the alias you use in your XML file to designate this namespace.

Of course the XML file is valid, nobody told otherwise. However, associated with the Groundspeak namespace are some semantics, which describe what each entity designates. cgeo has code which understands the meaning of a XML file using Groundspeak namespace, which it recognizes because of the URI. If you produce a XML file where the groundspeak namespace is associated with the `xmlns:spaceshuttle="http://www.groundspeak.com/cache/1/1"`, it will understand that `<spaceshuttle:cache/>` designates a Geocache.

Similarly, you could host the same schema file at a different URI and decide that now its `<cache/>` element designates a birthday cake.

If Geocaching Australia uses the same entities as GroundSpeak, it makes sense to use the existing URI. By duplicating the schema with a different URI, it forces applications which correctly use XML to also recognize this particular URI, and duplicate some code to act accordingly and recognize that `<geocachingaustralia:cache/>` also designates a Geocache. If everyone did that, XML namespaces would become quite useless, as for example anyone could host schemas from https://schema.org/ on their own sites and render them useless.

As for other applications, we cannot speak for them: either they have coded a special case for Geocaching Australia URI, or they only check the XML file against the schema and infer what the tag names might mean, or they might even only check the XML file for well-formedness and only look at the tag name (and maybe the prefix but ignoring what it really means).

**Comment by caughtatwork** [comment]
Thanks everyone. We've changed the XSD reference as suggested and all seems well. Appreciate your looking into this and offering solutions to fix.

**Comment by akkatracker** [comment]
Thanks @caughtatwork - any way for cgeo to have access to GCA data for live display of trigs and movables?
